src/devices/Servo/ServoMotor.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Device.Gpio;
+using System.Device.Pwm;
+using System.Device.Pwm.Drivers;
+
+namespace Iot.Device.Servo
+{
+    public class ServoMotor : IDisposable
+    {
+        private int servoPin;
+        private int pwmChannel;
+        private double PulseFrequency = 20.0;
+
+        double currentPulseWidth = 0;
+
+        private float angle;
+        /// <summary>
+        /// Servo motor definition.
+        /// </summary>
+        private ServoMotorDefinition definition;
+        /// <summary>
+        /// The duration per angle's degree.
+        /// </summary>
+        private double rangePerDegree;
+        private PwmController pwmController;
+
+        public ServoMotor(int pinNumber, int pwmChannel, ServoMotorDefinition definition)
+        {                       
+            this.definition = definition;
+            PulseFrequency = definition.Period / 1000.0;
+
+            UpdateRange();
+
+            servoPin = pinNumber;
+            this.pwmChannel = pwmChannel;
+            // try hardware PWM first
+            try{
+                pwmController = new PwmController();
+                pwmController.OpenChannel(pinNumber, pwmChannel);
+                Console.WriteLine("Using hardware PWM");
+
+            } catch
```

Comment:
might be worth to catch just exceptions we expect so that we can distinguish between real problem and hardware pwm not supported